paparazzi/src/main/java/app/cash/paparazzi/internal/apng/ApngReader.kt

```diff
@@ -0,0 +1,264 @@
+/*
+ * Copyright (C) 2023 Square, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package app.cash.paparazzi.internal.apng
+
+import app.cash.paparazzi.internal.apng.PngConstants.Header
+import app.cash.paparazzi.internal.apng.PngConstants.Header.ACTL
+import app.cash.paparazzi.internal.apng.PngConstants.Header.FCTL
+import app.cash.paparazzi.internal.apng.PngConstants.Header.FDAT
+import app.cash.paparazzi.internal.apng.PngConstants.Header.IDAT
+import app.cash.paparazzi.internal.apng.PngConstants.Header.IEND
+import app.cash.paparazzi.internal.apng.PngConstants.Header.IHDR
+import app.cash.paparazzi.internal.apng.PngConstants.PNG_COLOR_TYPE_RGB
+import app.cash.paparazzi.internal.apng.PngConstants.PNG_COLOR_TYPE_RGBA
+import app.cash.paparazzi.internal.apng.PngConstants.PNG_SIG
+import okio.Buffer
+import okio.BufferedSource
+import okio.FileSystem
+import okio.Path.Companion.toPath
+import okio.buffer
+import java.awt.image.BufferedImage
+import java.awt.image.ColorModel
+import java.awt.image.DataBufferInt
+import java.awt.image.Raster
+import java.io.Closeable
+import java.io.File
+import java.lang.IllegalStateException
+import java.util.zip.CRC32
+import java.util.zip.Inflater
+
+/**
+ * Provides a simple sub-spec implementation for reading APNG files. Only handles
+ * chunks which are implemented by the writer.
+ */
+internal class ApngReader(file: File) : Closeable {
+  private val crcEngine = CRC32()
+  private val fileHandle = FileSystem.SYSTEM.openReadOnly(file.path.toPath())
+  private val buffer = fileHandle.source().buffer()
```

Comment:
```suggestion
  private val buffer = Buffer()
```

What this does is it creates a `Source` object and immediately leaks it.

Reply:
Actually no it doesn’t, don’t mind me.